src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ApplyTransformationToGeometry.cpp

```diff
@@ -150,6 +148,21 @@ Result<> ApplyTransformationToGeometry::operator()()
     return MakeErrorResult(-84500, fmt::format("Keeping the original geometry is not supported."));
   }
 
+  auto* imageGeometryPtr = m_DataStructure.getDataAs<ImageGeom>(m_InputValues->SelectedGeometryPath);
+  const bool isNodeBased = (imageGeometryPtr == nullptr);
+
+  ImageRotationUtilities::Matrix4fR translationToGlobalOriginMat = ImageRotationUtilities::Matrix4fR::Identity();
+  ImageRotationUtilities::Matrix4fR translationFromGlobalOriginMat = ImageRotationUtilities::Matrix4fR::Identity();
+  if(isNodeBased)
+  {
+    Point3D<float32> minPoint = {0.0F, 0.0F, 0.0F};
+    auto& nodeGeometry0D = m_DataStructure.getDataRefAs<INodeGeometry0D>(m_InputValues->SelectedGeometryPath);
+    auto boundingBox = nodeGeometry0D.getBoundingBox();
+    minPoint = boundingBox.getMinPoint();
+    translationToGlobalOriginMat = ImageRotationUtilities::GenerateTranslationTransformationMatrix({-minPoint[0], -minPoint[1], -minPoint[2]});
+    translationFromGlobalOriginMat = ImageRotationUtilities::GenerateTranslationTransformationMatrix({minPoint[0], minPoint[1], minPoint[2]});
+  }
```

Comment:
```suggestion
  if(isNodeBased)
  {
    auto& nodeGeometry0D = m_DataStructure.getDataRefAs<INodeGeometry0D>(m_InputValues->SelectedGeometryPath);
    auto boundingBox = nodeGeometry0D.getBoundingBox();
    Point3Df minPoint = boundingBox.getMinPoint();
    translationToGlobalOriginMat = ImageRotationUtilities::GenerateTranslationTransformationMatrix({-minPoint[0], -minPoint[1], -minPoint[2]});
    translationFromGlobalOriginMat = ImageRotationUtilities::GenerateTranslationTransformationMatrix({minPoint[0], minPoint[1], minPoint[2]});
  }
```
just use move constructor to construct in place and use type alias